nssp/tests/test_patch.py

```diff
@@ -0,0 +1,130 @@
+import unittest
+from unittest.mock import patch as mock_patch, call
+from delphi_nssp.patch import patch, good_patch_config
+import os
+import shutil
+from datetime import datetime, timedelta
+
+class TestPatchModule(unittest.TestCase):
+    def test_good_patch_config(self):
+        # Case 1: missing custom_run flag and patch section
+        with mock_patch('logging.Logger') as mock_logger:
+            patch_config = {
+                "common": {}
+            }
+            self.assertFalse(good_patch_config(patch_config, mock_logger))
+            mock_logger.error.assert_has_calls([
+                call("Calling patch.py without custom_run flag set true."),
+                call("Custom flag is on, but patch section is missing."),
+            ])
+
+        # Case 2: missing end_issue in patch section
+        with mock_patch('logging.Logger') as mock_logger:
+            patch_config = {
+                "common": {
+                    "custom_run": True,
+                },
+                "patch": {
+                    "patch_dir": "dir",
+                    "start_issue": "2024-04-21",
+                    "source_dir": "source_dir"
+                }
+            }
+            self.assertFalse(good_patch_config(patch_config, mock_logger))
+            mock_logger.error.assert_has_calls([
+                call("Patch section is missing required key(s): end_issue"),
+            ])
+
+        # Case 3: start_issue not in yyyy-mm-dd format and source_dir doesn't exist.
+        with mock_patch('logging.Logger') as mock_logger:
+            patch_config = {
+                "common": {
+                    "custom_run": True,
+                },
+                "patch": {
+                    "patch_dir": "dir",
+                    "start_issue": "01-01-2024",
+                    "end_issue": "2024-04-22",
+                    "source_dir": "bad_source_dir"
+                }
+            }
+            self.assertFalse(good_patch_config(patch_config, mock_logger))
+            mock_logger.error.assert_has_calls([
+                call("Issue dates must be in YYYY-MM-DD format."),
+                call("Source directory bad_source_dir does not exist.")
+            ])
+
+        # Case 4: start_issue after end_issue
+        with mock_patch('logging.Logger') as mock_logger:
```

Comment:
can you break down the cases into individual tests? I find it easier to understand especially the test name is to test good config but you're also testing all the bad ones; 
the cases can be a part of the test names like test_bad_start_issue_config and etc (also helps when reading the build because it shows the individual tests)

Reply:
yup. added here [e29e07e](https://github.com/cmu-delphi/covidcast-indicators/pull/2000/commits/e29e07e715f3a3854f00fca89fa0568e4b7490ea)